test/test_e2e.rb

```diff
@@ -104,4 +109,27 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
```

Comment:
I disagree that we should couple this test the max requeues because

a. Fail fast has nothing to do with it
b. Changing the default behavior of the binary, that has no impact on the end result, can be confusing

As far as the speed is concerned, this is an e2e test (e.g. real world test), so IMO keeping things as close to default as possible is better

Reply:
Fair enough, though the test is always "coupled with the requeueing mechanism" since it's an e2e test, and requeues do affect the state of the queue (i.e. `processed_jobs`).